### ISSUE TITLE
ncp: not found ??

### ISSUE THREAD

**Issue by pauloresolve** [Automattic/wp-calypso#35673]
While trying "npm start" returned the following error:

\# npm start

> wp-calypso@0.17.0 prestart /var/www/calypso/wp-calypso
> npx check-node-version --package && npm run -s install-if-deps-outdated && node bin/welcome.js

npx: installed 13 in 3.212s
             _
    ___ __ _| |_   _ _ __  ___  ___
   / __/ _` | | | | | '_ \/ __|/ _ \
  | (_| (_| | | |_| | |_) \__ \ (_) |
   \___\__,_|_|\__, | .__/|___/\___/
               |___/|_|

> wp-calypso@0.17.0 start /var/www/calypso/wp-calypso
> npm run -s build

**sh: 1: ncp: not found**
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! wp-calypso@0.17.0 start: `npm run -s build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the wp-calypso@0.17.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-08-21T18_10_16_282Z-debug.log

This is the log:

0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@6.9.0
3 info using node@v10.16.3
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle wp-calypso@0.17.0~prestart: wp-calypso@0.17.0
6 verbose lifecycle wp-calypso@0.17.0~prestart: unsafe-perm in lifecycle true
7 verbose lifecycle wp-calypso@0.17.0~prestart: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/calypso/wp-calypso/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
8 verbose lifecycle wp-calypso@0.17.0~prestart: CWD: /var/www/calypso/wp-calypso
9 silly lifecycle wp-calypso@0.17.0~prestart: Args: [ '-c',
9 silly lifecycle   'npx check-node-version --package && npm run -s install-if-deps-outdated && node bin/welcome.js' ]
10 silly lifecycle wp-calypso@0.17.0~prestart: Returned: code: 0  signal: null
11 info lifecycle wp-calypso@0.17.0~start: wp-calypso@0.17.0
12 verbose lifecycle wp-calypso@0.17.0~start: unsafe-perm in lifecycle true
13 verbose lifecycle wp-calypso@0.17.0~start: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/calypso/wp-calypso/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
14 verbose lifecycle wp-calypso@0.17.0~start: CWD: /var/www/calypso/wp-calypso
15 silly lifecycle wp-calypso@0.17.0~start: Args: [ '-c', 'npm run -s build' ]
16 silly lifecycle wp-calypso@0.17.0~start: Returned: code: 1  signal: null
17 info lifecycle wp-calypso@0.17.0~start: Failed to exec start script
18 verbose stack Error: wp-calypso@0.17.0 start: `npm run -s build`
18 verbose stack Exit status 1
18 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
18 verbose stack     at EventEmitter.emit (events.js:198:13)
18 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
18 verbose stack     at ChildProcess.emit (events.js:198:13)
18 verbose stack     at maybeClose (internal/child_process.js:982:16)
18 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
19 verbose pkgid wp-calypso@0.17.0
20 verbose cwd /var/www/calypso/wp-calypso
21 verbose Linux 4.15.0-58-generic
22 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
23 verbose node v10.16.3
24 verbose npm  v6.9.0
25 error code ELIFECYCLE
26 error errno 1
27 error wp-calypso@0.17.0 start: `npm run -s build`
27 error Exit status 1
28 error Failed at the wp-calypso@0.17.0 start script.
28 error This is probably not a problem with npm. There is likely additional logging output above.
29 verbose exit [ 1, true ]

The package.json contains the dependency, in "/node_modules" there is a folder "ncp".

When trying "npm install ncp --no-optional --save" I get:

\# sudo npm install ncp --save
npm ERR! Cannot read property '0' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/viasonho/.npm/_logs/2019-08-21T18_27_39_362Z-debug.log

This is the log:

[...]
32155 verbose stack TypeError: Cannot read property '0' of undefined
32155 verbose stack     at rmStuff (/usr/local/lib/node_modules/npm/lib/unbuild.js:61:24)
32155 verbose stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
32155 verbose stack     at ret (eval at makeNodePromisifiedEval (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:13:39)
32155 verbose stack     at lifecycle.then.then (/usr/local/lib/node_modules/npm/lib/install/action/unbuild.js:12:12)
32155 verbose stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
32155 verbose stack     at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
32155 verbose stack     at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)
32155 verbose stack     at Promise._settlePromise0 (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
32155 verbose stack     at Promise._settlePromises (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:694:18)
32155 verbose stack     at Promise._fulfill (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:638:18)
32155 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/nodeback.js:42:21
32156 verbose cwd /var/www/calypso/wp-calypso
32157 verbose Linux 4.15.0-58-generic
32158 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "ncp" "--save"
32159 verbose node v10.16.3
32160 verbose npm  v6.9.0
32161 error Cannot read property '0' of undefined
32162 verbose exit [ 1, true ]

My versions:
node v10.16.3
npm v6.9.0
My system:
Ubuntu server 19.04 LTS

Installation:
https://wpcalypso.wordpress.com/devdocs/docs/install.md

I suspect the problem is with uninstall. Because the installation stops at uninstall: undefined

How to solve?

**Comment by pauloresolve** [comment]
My versions:
node v10.16.3
npm v6.9.0
My system:
Ubuntu server 19.04 LTS

Installation:
https://wpcalypso.wordpress.com/devdocs/docs/install.md

**Comment by pauloresolve** [comment]
I suspect the problem is with uninstall. Because the installation stops at uninstall: undefined

**Comment by victorespigares** [comment]
I was experiencing the same `ncp: not found` error on my wp-calypso-bootstrap instance. 

After some fumbling, I finally got it to work again. Here's what I did, in case it helps you pinpoint your issue:

- Removed `node_modules` folder (actually I moved them to another folder, just in case).
- Ran `npm run clean`
- Did a `git pull` to get the latest from the repo
- Ran `npm install`
- Ran `npm start` again... and it worked :)

Not sure which one did it, but I suspect it was a combination of old cached dependencies or something like that, that wasn't working for me.

Hope this helps!
